src/app/core/services/issue.service.ts

```diff
@@ -85,24 +85,22 @@ export class IssueService {
     );
   }
 
-  /**
-   * This function will update the issue's state of the application. This function needs to be called whenever a issue is deleted.
-   */
-  deleteFromLocalStore(issueToDelete: Issue) {
-    const { [issueToDelete.id]: issueToRemove, ...withoutIssueToRemove } = this.issues;
-    this.issues = withoutIssueToRemove;
-    this.issues$.next(Object.values(this.issues));
-  }
-
   /**
    * This function will update the issue's state of the application. This function needs to be called whenever a issue is added/updated.
+   *
+   * @params issuesToUpdate - An array of issues to update the state of the application with.
+   * @params shouldEmit - Whether the updated issues should be emitted to issues$.
    */
-  updateLocalStore(issueToUpdate: Issue) {
-    this.issues = {
-      ...this.issues,
-      [issueToUpdate.id]: issueToUpdate
-    };
-    this.issues$.next(Object.values(this.issues));
+  updateLocalStore(issuesToUpdate: Issue[], shouldEmit: boolean = true) {
+    const newIssues = { ...this.issues };
+    issuesToUpdate.forEach((issue) => {
```

Comment:
This method is only in use within the class, and the method that is referencing it is also a private method (`createAndSaveIssueModels`). Let's make this a private method as well.